airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/operation/DefaultSyncOperation.kt

```diff
@@ -102,7 +104,20 @@ class DefaultSyncOperation<DestinationState : MinimumDestinationState>(
     override fun flushStream(descriptor: StreamDescriptor, stream: Stream<PartialAirbyteMessage>) {
         val streamConfig =
             parsedCatalog.getStream(descriptor.namespace ?: defaultNamespace, descriptor.name)
-        streamOpsMap[streamConfig.id]?.writeRecords(streamConfig, stream)
+        streamOpsMap[streamConfig.id]?.writeRecords(
+            streamConfig,
+            stream.map { record ->
+                if (record.record!!.meta == null) {
```

Comment:
do we really want to modify the record object? That's a lot of `!!`!!

Reply:
I think modifying the object makes sense, most destinations are just going to `Jsons.serialize(record.record!!.meta)`, so this is the best place to have logic that's identical across destinations. Otherwise we'd need to inject syncId down through multiple layers, just so that each implementation can do `Jsons.serialize(meta.add("sync_id", syncId))`

though I have some opinions about
* passing around AirbyteMessage instead of AirbyteRecordMessage
* directly using the protocol object (i.e. why are these fields nullable to begin with)